DataConservancy/pass-ember
# Initial submission creation workflow

## Overview
* Adds DOI -> Details -> View workflow for creating a new submission
* Persists workflow to repository
* Uses DOI service for resolving DOI metadata
* Transitions to next workflow stage (funding) when approopriate

## Notable omissions:
* There is nothing related to journal in this step yet; that is to be completed as per #18 

## How To Test
* Check out this PR locally 
  1.  Check out the latest `master` from pass-ember (e.g. `git checkout master` then `git pull`)
  2.  Create a branch from master to try it out in : `git checkout -b birkland-issue-15 master`
  3.  Pull the PR into it `git pull https://github.com/birkland/pass-ember.git issue-15`
  4.  Do `npm install` and then `ember serve`
* From grants table
  1. Click on an entry in the `award number` or `submissions` column to see grant details
  2. At the bottom, click `new submission`, a modal dialog box should pop up.  Interact with it
      * Try canceling and see what that does
      * Type in a legitimate DOI and click next - the title should be pre-populated
         * Click _cancel_ - the dialog should go away, and you can start anew
         * Click _save_ - The dialog box should go away, you should see the new submission entry in the table
    3. Fill out a DOI and title, but this time click _save and continue_, you should be directed to the submission page and see an overview of what you just typed in, with an option to advance to the next step
* From the submissions table
  1. Click on the _create new submission_ button
  2. Enter in DOI (or not), click 'next`.  Enter in title (or not, if it's pre-populated)
      * If you click _cancel_, you should go back to the submissions table
      * If you click _save and continue_, you should be directed to the submissions page

Resolves #15

Interested parties: @htpvu
